### PR TITLE
Handling and Throwing

### PR DESCRIPTION
**Pull request by GleasonC4**
Addresses #91 
- Does not include function exception throwing
- Full implementation of try/catch/finally and throwing
- Native exceptions will follow in next update

### CODE REVIEW

File: test/end-to-end/Exception Handling/exception handling.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-
+try {
+    // ...
+}
```
Comment (JoshuaKGoldberg) [inline test/end-to-end/Exception Handling/exception handling.ts]: I believe the standard for TypeScript would be to put `}` and continuations on the same line:

```typescript
try {
    // ...
} catch (error) {
    // ...
}

File: test/end-to-end/Exception Handling/exception handling.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+catch (exceptionType foo) {
```
Comment (JoshuaKGoldberg) [inline test/end-to-end/Exception Handling/exception handling.ts]: This is invalid TypeScript. You can always validate using http://www.typescriptlang.org/play/index.html.

File: test/end-to-end/Exception Handling/exception handling.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+-
+try {
+    // ...
+}
+catch (exceptionType foo) {
```
Comment (JoshuaKGoldberg) [inline test/end-to-end/Exception Handling/exception handling.java]: As with TypeScript, I believe this should be on the same line.

File: test/end-to-end/Exception Handling/exception handling.gls [hunk 4]
```diff
@@ -0,0 +1,11 @@
+-
+try start
+    comment line : ...
+try end
+catch start : exceptionType foo
```
Comment (JoshuaKGoldberg) [inline test/end-to-end/Exception Handling/exception handling.gls]: Again, PascalCase for the exception types.

File: test/end-to-end/Exception Handling/exception handling.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+-
+try:
+    # ...
+
```
Comment (JoshuaKGoldberg) [inline test/end-to-end/Exception Handling/exception handling.py]: Is this blank line standard per Python style guidelines? I thought it wasn't, but am not very familiar.

File: src/Languages/PythonicLanguage.ts [hunk 6]
```diff
@@ -121,6 +131,8 @@ export abstract class PythonicLanguage extends Language {
     protected generateFunctionProperties(functions: FunctionProperties): void {
         functions.defineStartLeft = "def ";
         functions.defineStartRight = ":";
+
```
Comment (JoshuaKGoldberg) [inline src/Languages/PythonicLanguage.ts]: Nit: Why the blank line?

File: src/Commands/CatchStartCommand.ts [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+/**
+ * A command for the beginning of a catch block.
+ */
+export class CatchStartCommand extends Command {
+    /**
+     * Information on parameters this command takes in.
+     */
+    private static parameters: Parameter[] = [
+        new SingleParameter("exception", "Target exception.", true),
+        new SingleParameter("alias", "Alias for target exception.", true)
```
Comment (JoshuaKGoldberg) [inline src/Commands/CatchStartCommand.ts]: More accurately, this would be a single parameter for the exception. Must all exception types require one?

Comment (GleasonC4) [inline src/Commands/CatchStartCommand.ts]: The "alias", as far as I can tell, is required for TypeScript, and therefore i thought it should be required for GLS. Correct me if i'm wrong on that.

Comment (JoshuaKGoldberg) [inline src/Commands/CatchStartCommand.ts]: I was thinking of the wrong part of this - you're right.

**Pull request merged**